src/app/charts/core/templates/_helpers-sso.tpl

```diff
@@ -164,12 +164,13 @@ idp.xml
       {{- /* The client is valid! Clean the map a little */ -}}
       {{- $client = (omit $client "enabled") -}}
 
-      {{- /* Set the usersDirectory to the specified value */ -}}
-      {{- $client = set $client "usersDirectory" $usersDirectory -}}
-
       {{- /* Set the redirectUri to the specified value */ -}}
       {{- $client = set $client "redirectUri" (printf "%s/%s" $baseUrl $client.registrationId) -}}
 
+      {{- $usersDirectory := ((hasKey $client "usersDirectory") | ternary $client.usersDirectory $defaultUsersDirectory) -}}
+      {{- $client = set $client "usersDirectory" $usersDirectory -}}
+      
```

Comment:
Please don't add empty lines with leading spaces :D

If we're going to demand an end to sloppiness, it needs to start with us (to the best of our abilities)